File: ScaleformUI_Csharp/Menus/Pause Menus/Elements/Items/MinimapRaceCheckpoint.cs
using CitizenFX.Core;

namespace ScaleformUI.PauseMenus.Elements.Items
{
    public class MinimapRaceCheckpoint
    {
        public Vector3 Position;
        public int BlipSprite;
        public float Scale = 0;
        public BlipColor Color = BlipColor.White;
        public bool Number;

        public MinimapRaceCheckpoint() { }
        public MinimapRaceCheckpoint(Vector3 position, int blipSprite)
        {
            Position = position;
            BlipSprite = blipSprite;
        }
        public MinimapRaceCheckpoint(BlipSprite sprite, Vector3 position)
        {
            BlipSprite = (int)sprite;
            Position = position;
        }
    }
}

File: ScaleformUI_Lua/src/Menus/PauseMenus/Elements/Items/MinimapRaceCheckpoint.lua
MinimapRaceCheckpoint = setmetatable({}, MinimapRaceCheckpoint)
MinimapRaceCheckpoint.__index = MinimapRaceCheckpoint
MinimapRaceCheckpoint.__call = function()
    return "LobbyItem", "MinimapRaceCheckpoint"
end

---@class MinimapRaceCheckpoint
---@field private New fun(sprite:number, position:vector3, color:HudColours, scale:number, number:number|boolean)
---@field public Sprite number
---@field public Position vector3

function MinimapRaceCheckpoint.New(sprite, position, color, scale, number)
    local _data = {
        Sprite = sprite,
        Position = position,
        Scale = scale or 0.0,
        Color = color or HudColours.HUD_COLOUR_WHITE,
        Number = number,
    }
    return setmetatable(_data, MinimapRaceCheckpoint)
end
